app/jobs/deposit_collection_job.rb

```diff
@@ -4,9 +4,12 @@
 class DepositCollectionJob < ApplicationJob
   # @param [CollectionForm] collection_form
   # @param [Collection] collection
-  def perform(collection_form:, collection:)
+  # @param [User] current_user
+  def perform(collection_form:, collection:, current_user:)
     @collection_form = collection_form
     @collection = collection
+    # Setting current user so that it will be available for notifications.
+    Current.user = current_user
```

Comment:
This was working locally because jobs were being run in the same process; however, in production they're not so need to set the current user.